backend/frontend/views.py

```diff
@@ -650,6 +651,15 @@ def view_partial_teams_list(request):
             user=user, perms=Team.PERMISSION_VIEW, klass=all_teams, any_perm=True
         )
 
+    if request.method == "POST":
```

Comment:
The algorithm here is weird:

1. get the teams that the user is allowed to view
2. if there is a post then totally ignore all the work we just did and get a new list of teams that doesn't take permissions into account at all

Better would be:

1. Get the teams
2. if the user is tiltering by name, filter the list by name
3. filter out what the user is not allowed to see 

So we should have one queryset and we should refine it over time instead of just starting over.

Right now any user can just filter by "c" to get a full list of cohorts.

Reply:
I instead updated this to be:
1. Get the teams
2. Filter the teams the user is allowed to see
3. Use the teams the user is allowed to see when filtering via search term